src/frontend/src/pages/PIADetailPage/index.tsx

```diff
@@ -0,0 +1,166 @@
+import {
+  IPIAIntake,
+  IPIAIntakeResponse,
+} from '../../types/interfaces/pia-intake.interface';
+import { faArrowDown } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { dateToString } from '../../utils/date';
+import messages from './messages';
+
+import { useNavigate, useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { HttpRequest } from '../../utils/http-request.util';
+import { API_ROUTES } from '../../constant/apiRoutes';
+import { routes } from '../../constant/routes';
+
+const PIADetailPage = () => {
+  // https://github.com/microsoft/TypeScript/issues/48949
+  // workaround
+  const win: Window = window;
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const [pia, setPia] = useState<any>({});
+  useEffect(() => {
+    (async () => {
+      try {
+        // Actually perform fetch
+        const result = (
+          await HttpRequest.get<IPIAIntakeResponse>(
+            API_ROUTES.GET_PIA_INTAKE.replace(':id', `${id}`),
+          )
+        ).data;
+        setPia(result);
+      } catch (e: any) {
+        const errorCode: string = e.message.split(':')[1];
```

Comment:
I believe @kushal-arora-fw had some comment doing it this way. If the error message changes, thing might break the string? 
Also, referencing index [1] is prone to break long run.

I personally might add a comment giving an example of the actual error message. This way devs might get an understanding what index 1 looks like.

Reply:
it will update in my pr as well